src/@next/CurrencyInput/CurrencyInput.tsx

```diff
@@ -47,21 +56,50 @@ export const CurrencyInput = React.forwardRef<
     return Number.isNaN(cleanedValue) ? 0 : cleanedValue;
   };
 
+  const getCurrencySymbol = (locale: string, currencyCode: string) => {
+    const formatter = new Intl.NumberFormat(locale, {
+      style: 'currency',
+      currency: currencyCode,
+    });
+
+    const parts = formatter.formatToParts(0);
+    const currencySymbol = parts.find(part => part.type === 'currency')?.value;
+
+    return currencySymbol;
+  };
+
   const [formattedValue, setFormattedValue] = useState(
     formatter.format(getRawNumber(value.toString()))
   );
 
+  let prefix: string;
+
+  if (currencySymbol) {
```

Comment:
can we put all this to `getCurrencySymbol` function instead with early return when condition fulfilled?

Reply:
Updated